extensions/machine-learning/src/packageManagement/sqlPythonPackageManageProvider.ts

```diff
@@ -67,13 +67,14 @@ export class SqlPythonPackageManageProvider extends SqlPackageManageProviderBase
 			let port = '1433';
 			let server = connection.serverName;
 			let database = databaseName ? `, database="${databaseName}"` : '';
+			const auth = connection.userName ? `, uid="${connection.userName}", pwd="${credentials[azdata.ConnectionOptionSpecialType.password]}"` : '';
 			let index = connection.serverName.indexOf(',');
 			if (index > 0) {
 				port = connection.serverName.substring(index + 1);
 				server = connection.serverName.substring(0, index);
 			}
 
-			let pythonConnectionParts = `server="${server}", port=${port}, uid="${connection.userName}", pwd="${credentials[azdata.ConnectionOptionSpecialType.password]}"${database})`;
+			let pythonConnectionParts = `server="${server}", port=${port}${auth}${database})`;
```

Comment:
as this slightly expands in complexity, maybe it makes sense to switch over to adding the pieces to an array, then `string.join(', ')`ing them into the final `pythonConnectionParts` string?  Doing this sorta stuff via string concat feels like a yellow flag to me

Reply:
you're right, have to merge this for June release soon but will refactor later